evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +123,44 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByTypeorTag(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type or Tag Transformer"
```

Comment:
Just food for thought, but if we want to allow users to use this transformer via the name, this is a mighty long name that they'd have to not make a typo for... lolol

Reply:
Edit: I like @dsherry's suggestion of `SelectByType` :)

Reply:
Good point @angela97lin, I didn't think about that! Do you think just "Select By Type Transformer" would be concise but also informative enough?

Reply:
Yup, I think that's great 😁